### PR TITLE
use depguard to prohibitate importing github.com/golang/protobuf

### LINKED ISSUE #868: use depguard to prohibitate importing `github.com/golang/protobuf`
**Task description**:

`github.com/golang/protobuf` has been supseded by `google.golang.org/protobuf`, which means almost all the functions are equavaliently implemented there.

But in kmesh we are mixing import anyone of them.  [depguard](https://golangci-lint.run/usage/linters/#depguard) provides a good way to set allowed importing packages. We can make use of it to not allow `github.com/golang/protobuf`

**Solution**:

**Who can join or take the task**:

The good first issue is intended for `first-time contributors` to get started on his/her contributor journey.

After a contributor has successfully completed 1-2 good first issue's,
they should be ready to move on to `help wanted` items, saving the remaining `good first issue` for other new contributors.

**How to join or take the task**:

Just reply on the issue with the message `/assign` in a separate line.

Then, the issue will be assigned to you.

**How to ask for help**:

If you need help or have questions, please feel free to ask on this issue.
The issue author or other members of the community will guide you through the contribution process.

### PR DESCRIPTION
**Pull request by SpongeBob0318**
**What type of PR is this?**
/kind enhancement

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #868

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: common/config/.golangci.yaml [hunk 1]
```diff
@@ -49,6 +50,18 @@ linters-settings:
   gocyclo:
     # minimal cyclomatic complexity to report
     min-complexity: 15
+  depguard:
+    rules:
+      main:
+        allow:
```
Comment (hzxuzhonghu) [inline common/config/.golangci.yaml:56]: seems you should remove the allow, otherwise it will treat unlisted package disallowed

**Comment by SpongeBob0318** [comment]
@hzxuzhonghu Sorry, this is my first time doing open source and I'm a bit unfamiliar with it. Do I still need to modify the content of import “GitHub/golang/protobuf”, such as in cache_test.go, after removing “allow”

File: Makefile [hunk 2]
```diff
@@ -117,6 +117,8 @@ install:
 	$(QUIET) make install -C bpf/deserialization_to_bpf_map
 	$(QUIET) make install -C kernel/ko_src
 
+	$(QUIET) go install github.com/OpenPeeDeeP/depguard/cmd/depguard@latest
```
Comment (zirain) [inline Makefile:120]: this seems not a must?

Comment (SpongeBob0318) [reply Makefile:120]: thank you, i will remove it

File: common/config/.golangci.yaml [hunk 3]
```diff
@@ -49,6 +50,15 @@ linters-settings:
   gocyclo:
     # minimal cyclomatic complexity to report
     min-complexity: 15
+  depguard:
+    rules:
+      main:
+        deny:
+          - pkg: "github.com/golang/protobuf"
+            desc: "Please use google.golang.org/protobuf instead."
+          - pkg: 'gopkg.in/yaml.v3'
```
Comment (zirain) [inline common/config/.golangci.yaml:59]: `gopkg.in/yaml.v2` and `github.com/ghodss/yaml` maybe import by accident, you should add them too.

Comment (SpongeBob0318) [reply common/config/.golangci.yaml:59]: of course, i will solve it soon

File: pkg/controller/workload/workload_hash.go [hunk 4]
```diff
@@ -23,7 +23,8 @@ import (
 	"math"
 	"os"
 
-	"gopkg.in/yaml.v3"
+	//"gopkg.in/yaml.v3"
```
Comment (hzxuzhonghu) [inline pkg/controller/workload/workload_hash.go:26]: remove

**Pull request closed without merging**